autoPyTorch/pipeline/components/preprocessing/tabular_preprocessing/feature_preprocessing/base_feature_preprocessor_choice.py

```diff
@@ -0,0 +1,110 @@
+        """
+        components: Dict = OrderedDict()
+        components.update(_preprocessors)
+        components.update(_addons.components)
+        return components
+
+    def get_hyperparameter_search_space(self,
+                                        dataset_properties: Optional[Dict[str, Any]] = None,
+                                        default: Optional[str] = None,
+                                        include: Optional[List[str]] = None,
+                                        exclude: Optional[List[str]] = None) -> ConfigurationSpace:
+        cs = ConfigurationSpace()
+
+        if dataset_properties is None:
+            dataset_properties = dict()
+
+        dataset_properties = {**self.dataset_properties, **dataset_properties}
+
+        available_ = self.get_available_components(dataset_properties=dataset_properties,
+                                                   include=include,
+                                                   exclude=exclude)
+
+        if len(available_) == 0:
+            raise ValueError("no feature preprocessors found, please add a feature preprocessor")
+
+        if default is None:
+            defaults = ['NoFeaturePreprocessor',
+                        'FastICA',
+                        'KernelPCA',
+                        'RandomKitchenSinks',
+                        'Nystroem',
+                        'PolynomialFeatures',
+                        'PowerTransformer',
+                        'TruncatedSVD',
+                        ]
+            for default_ in defaults:
+                if default_ in available_:
+                    if include is not None and default_ not in include:
+                        continue
+                    if exclude is not None and default_ in exclude:
+                        continue
+                    default = default_
+                    break
+
+        # add only no feature preprocessor to choice hyperparameters in case the dataset is only categorical
+        if len(dataset_properties['numerical_columns']) == 0:
+            default = 'NoFeaturePreprocessor'
+            preprocessor = CSH.CategoricalHyperparameter('__choice__',
+                                                         ['NoFeaturePreprocessor'],
+                                                         default_value=default)
+        else:
+            preprocessor = CSH.CategoricalHyperparameter('__choice__',
+                                                         list(available_.keys()),
+                                                         default_value=default)
+
+        cs.add_hyperparameter(preprocessor)
+
+        # add only child hyperparameters of early_preprocessor choices
+        for name in preprocessor.choices:
+            updates = self._get_search_space_updates(prefix=name)
+            config_space = available_[name].get_hyperparameter_search_space(dataset_properties,  # type:ignore
```

Comment:
why type ignore here?

Reply:
so as get_hyperparameter_search_space of the base component only has dataset properties as a parameter, it gives a "too many arguments for get_hyperparameter search space" error. If I add a **kwargs to the base component, it gives signature invalid for every class inheriting from base component. On searching on google, I found [this](https://github.com/python/mypy/issues/5364)